### PR TITLE
Replace net451/log4net with .NET Standard/Core logging

### PR DESCRIPTION
**Pull request by DaveAurionix**
Replace log4net with .NET Core Logging, removing support for net451 in the process.  Use .AddDefaultPatLogger (.NET Core IoC) or .WithDefaultPatLogger (StructureMap IoC)

### CODE REVIEW

**Comment by ilivewithian** [comment]
@DaveAurionix what was the driver to remove support for .net 451? IIRC we still need to support it.

**Comment by DaveAurionix** [comment]
@AndyRB and I had a chat on the #pat Slack channel.  Apparently, from PB's angle, Subscribers don't need .NET 451 but Publishers still do.  You think that might not be accurate?

**Comment by DaveAurionix** [comment]
@ilivewithian See the thread off this: https://opensource-pbtech.slack.com/archives/CBQGJKP8X/p1532082636000280

**Comment by DaveAurionix** [comment]
Still supports net47 BTW, just not earlier versions including net451.

File: Pat.Subscriber.IntegrationTests/DependencyResolution/DotNetIoC.cs [hunk 1]
```diff
@@ -52,9 +51,10 @@ public static IServiceCollection Initialize(IConfigurationRoot configuration)
                         provider.GetRequiredService<IMessageSender>(),
                         provider.GetRequiredService<IMessageGenerator>(),
                         new MessageProperties(Guid.NewGuid().ToString())))
-                .AddSingleton(LogManager.GetLogger(loggerName, loggerName))
                 .AddTransient<IMessageSender, MessageSender>()
                 .AddTransient<IStatisticsReporter, StatisticsReporter>()
+                .AddDefaultPatLogger()
+                .AddLogging( b=> b.AddDebug())
```
Comment (AndyRB) [inline Pat.Subscriber.IntegrationTests/DependencyResolution/DotNetIoC.cs]: formatting

**Pull request merged**